tests/periph_backup_ram/main.c

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2019 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup tests
+ * @{
+ *
+ * @file
+ * @brief       Low-level test for backup RAM
+ *
+ * @author      Benjamin Valentin <benjamin.valentin@ml-pa.com>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+
+#include "periph/rtc.h"
+#include "pm_layered.h"
+#include "xtimer.h"
+
+#ifndef SLEEP_SEC
+#define SLEEP_SEC 1
+#endif
+
+int main(void)
+{
+    /* We keep a copy of counter in the .noinit section (normal RAM).
+     * In the deepest sleep normal RAM is not retained, so if it matches
+     * counter anyway after wakeup, we did not sleep properly.
+     */
+    static int counter_noinit __attribute__((section(".noinit")));
+    static int counter __attribute__((section(".backup.bss")));
+
+    if (counter == 0) {
+        puts("\nBackup RAM test\n");
+        printf("This test will increment the counter by 1, "
+               "then enter deep sleep for %ds\n", SLEEP_SEC);
+    } else if (counter_noinit == counter) {
+        puts("WARNING: non-backup memory retained - did we really enter deep sleep?");
+    }
+
+    printf("counter: %d\n", ++counter);
+    counter_noinit = counter;
+
+    /* Some tools have trouble flashing MCUs in deep sleep.
+     * Wait a bit to make re-flashing / debugging easier.
+     */
+    xtimer_sleep(3);
```

Comment:
This traditional delay is troublesome because you're suppose to see the counter increments by one every second but currently this is not the case.

Reply:
> you're suppose to see the counter increments by one every second 

Says who? :wink: 
I've added some output to clarify the delay periods.

It's really annoying when you have OpenOCD and your MCU enters deep sleep with no delay - there is no way to re-flash it. (If you're lucky you have JLink or edbg. But I'd rather save the unsuspecting Raspberry Pi GPIO users the cursing.)

Reply:
I don't mind keeping this delay but at least update the printf so users wont think this test is too slow ;)_